### ISSUE TITLE
🐛 [WOTLK] Six Demon Bag

### ISSUE THREAD

**Issue by al3xc1985** [cmangos/issues#3476]
### Bug Details

https://www.wowhead.com/classic/item=7734/six-demon-bag

The problem is that when u get the chance on using item to polymorph, the polymorph spell is cast on you and not on enemy.

It should be casted on enemy

### Steps to Reproduce

1.
2.
3.
4.

### Core SHA1 Commit Hash

4aa81134be302fed4a22f1c2e01374e3c3

### Database SHA1 Commit Hash

87d213957c5f6d4e2e3679eaf81986eb43e32525

### Operating System

windows 10

### Client Version

3.3.5a (Wrath of the Lich King)

**Comment by evil-at-wow** [comment]
> The problem is that when u get the chance on using item to polymorph, the polymorph spell is cast on you and not on enemy.
> It should be casted on enemy

I have no first hand experience, but it looks like both are possible: it can polymorph the enemy, but also yourself. See for example [this comment](https://www.wowhead.com/tbc/spell=14537/six-demon-bag#comments:id=2775671) and many other similar comments on the internet.

**Comment by al3xc1985** [comment]
I agree with that. But from ALL the tests I did, it never polymorphed the enemy, only me

**Comment by al3xc1985** [comment]
it might be just me. I will keep testing maybe I get the other effect 2

**Comment by al3xc1985** [comment]
Tested on deadmines, on a large number or mobs, and the result is what i wronte in description. No polymorph on enemy , only on you

**Comment by virusav** [comment]
Core SHA: 7c72208cd373e642f60732656182f83259fc45ea.
Wotlk-db SHA: a5e9d2890da486481b4b0f9bae51ecb1f0535343.

Patch:
```DIFF
diff --git a/src/game/Spells/SpellEffects.cpp b/src/game/Spells/SpellEffects.cpp
index e10645414..e7673e5bf 100644
--- a/src/game/Spells/SpellEffects.cpp
+++ b/src/game/Spells/SpellEffects.cpp
@@ -961,7 +961,7 @@ void Spell::EffectDummy(SpellEffectIndex eff_idx)
                         else if (roll < 90)                     // Summon Felhund minion (5% chance)
                             m_caster->CastSpell(m_caster, 14642, TRIGGERED_OLD_TRIGGERED, m_CastItem);
                         else if (roll < 97)                     // Polymorph (7% to target, 10% chance total)
-                            m_caster->CastSpell(unitTarget, 14621, TRIGGERED_OLD_TRIGGERED, m_CastItem);
+                            unitTarget->CastSpell(unitTarget, 14621, TRIGGERED_OLD_TRIGGERED, m_CastItem);
                         else                                    // Polymorph (3% to self, backfire, 10% chance total)
                             m_caster->CastSpell(m_caster, 14621, TRIGGERED_OLD_TRIGGERED, m_CastItem);
                     }

```
Spell 14621 - TARGET_SELF, so the target must cast the spell (or you need to somehow redefine the target through casting parameters).

With the patch, polymorph can be applied to a target or a caster.

The only strange thing is that when casting a “long” spell (of the possible effects of the spell 14537), there is no animation of casting the spell, i.e. the player simply stands for 2-3 seconds, and then the instant cast effect occurs.
